src/OpenFlight/Users/Application/UserLogin.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+
+namespace CodelyTv\OpenFlight\Users\Application;
+
+
+use CodelyTv\OpenFlight\Users\Domain\UserRepository;
+
+class UserLogin
+{
+
+    /**
+     * UserLogin constructor.
+     * @param UserRepository $repository
+     */
+    public function __construct(private UserRepository $repository)
+    {
+    }
+
+    public function __invoke(string $username, string $password): void
+    {
+        $user = $this->repository->findByUsername($username);
+        $user->checkPassword($password);
```

Comment:
Aqui hay un leak de dominio en la capa de aplicación. 
Le están diciendo a su dominio `COMO` tiene que hacer las cosas (`checkPassword`) en vez de decirle QUE hacer (`Login`). 
[Patrón Tell don't ask](https://martinfowler.com/bliki/TellDontAsk.html). Esto quiere decir que no le digamos a nuestro dominio que para hacer un Login (nuestro caso de uso), tiene que hacer un `checkPassword`. Que pasaría si el día de mañana se tienen que hacer otras comprobaciones sobre el dominio? La capa de aplicación tendría que ser consiente y hacer otras comprobaciones y eso esta mal por que es un leak de dominio en aplicación.
```suggestion
        $user->Login($password);
```